src/writer.rs

```diff
@@ -94,23 +93,23 @@ fn version_to_version_string(version: GpxVersion) -> GpxResult<&'static str> {
     match version {
         GpxVersion::Gpx10 => Ok("1.0"),
         GpxVersion::Gpx11 => Ok("1.1"),
-        version => Err(GpxError::UnknownVersionError(version)),
+        version @ GpxVersion::Unknown => Err(GpxError::UnknownVersionError(version)),
     }
 }
 
 fn version_to_xml_url(version: GpxVersion) -> GpxResult<&'static str> {
     match version {
-        GpxVersion::Gpx10 => Ok("http://www.topografix.com/GPX/1/0"),
-        GpxVersion::Gpx11 => Ok("http://www.topografix.com/GPX/1/1"),
-        version => Err(GpxError::UnknownVersionError(version)),
+        GpxVersion::Gpx10 => Ok("https://www.topografix.com/GPX/1/0"),
```

Comment:
Let's not do this, that URL identifies the schema, and it's not meant to be browsed. See the `targetNamespace` in https://www.topografix.com/GPX/1/1/gpx.xsd.